backend/windmill-api/src/jobs.rs

```diff
@@ -6526,8 +6730,74 @@ fn get_job_update_sse_stream(
             }
         }
     });
+}
 
-    tokio_stream::wrappers::ReceiverStream::new(rx)
+async fn get_flow_stream_delta(
+    db: &DB,
+    flow_status: &Option<sqlx::types::Json<Box<RawValue>>>,
+    flow_stream_job_id: &mut Option<Uuid>,
+    stream_offset: Option<i32>,
+) -> error::Result<Option<(Option<String>, Option<i32>)>> {
+    let job_id = if let Some(job_id) = flow_stream_job_id {
+        Some(*job_id)
+    } else if let Some(sqlx::types::Json(flow_status)) = flow_status {
+        let flow_status = serde_json::from_str::<FlowStatus>(flow_status.get())?;
+        match flow_status.modules.last() {
+            Some(FlowStatusModule::InProgress { is_stream: Some(true), job, .. }) => {
+                *flow_stream_job_id = Some(*job);
+                Some(*job)
+            }
+            Some(FlowStatusModule::InProgress { branch_chosen: Some(_), job, .. }) => {
```

Comment:
don't think it will work. What about branchone of branchone ? What about setting the is_stream complexity on the write side?